media_driver/agnostic/common/vp/hal/vphal_renderer.cpp

```diff
@@ -1427,6 +1427,7 @@ MOS_STATUS VphalRenderer::Initialize(
     eStatus = MOS_STATUS_SUCCESS;
 
 finish:
+    MOS_SafeFreeMemory(pKernelBin);
```

Comment:
this is also dangerous, it modifies the original logic.
even eStatus is success, the memory would be freed. Is that expected??